metrics/metrics.go

```diff
@@ -60,20 +60,40 @@ func NewMetrics(procName string) *Metrics {
 
 	return &Metrics{
 		Up: factory.NewGauge(prometheus.GaugeOpts{
-			Namespace: ns,
+			Namespace: namespace,
+			Subsystem: subsystem,
 			Name:      "up",
 			Help:      "1 if the proxy server has finished starting up",
 		}),
 		Info: factory.NewGaugeVec(prometheus.GaugeOpts{
-			Namespace: ns,
+			Namespace: namespace,
+			Subsystem: subsystem,
 			Name:      "info",
 			Help:      "Pseudo-metric tracking version and config info",
 		}, []string{
 			"version",
 		}),
-		RPCMetrics: metrics.MakeRPCMetrics(ns, factory),
-		registry:   registry,
-		factory:    factory,
+		HTTPServerRequestsTotal: factory.NewCounterVec(prometheus.CounterOpts{
+			Namespace: namespace,
+			Subsystem: httpServerSubsystem,
+			Name:      "requests_total",
+			Help:      "Total requests to the HTTP server",
+		}, []string{
+			"method", "status",
+		}),
+		HTTPServerRequestDurationSeconds: factory.NewHistogramVec(prometheus.HistogramOpts{
+			Namespace: namespace,
+			Subsystem: httpServerSubsystem,
+			Name:      "request_duration_seconds",
+			// TODO: we might want different buckets for different routes?
```

Comment:
lets file an issue for this within the repo

Reply:
within optimism repo you mean?
like a feature to pass your own buckets? Possibly, although their metrics server is also for rpc based requests, not http. So it just felt very different when I was reading it, as opposed to a typical REST endpoint which we have here.
Think its better to just fork and use our own stuff, we'll prob add a bunch of other things.

Reply:
I mean our repo :) 
just so we can track

Reply:
Ah gotcha! https://github.com/Layr-Labs/eigenda-proxy/issues/96